lib/xmalloc.c

```diff
@@ -340,6 +340,7 @@ xrealloc_impl(void *ptr, size_t new_size, const char *file, int line,
   new_ptr = realloc(ptr, new_size);
```

Comment:
Here, we attempt to allocate a new object with the new size. If successful, the old object will be freed.